### ISSUE TITLE
Feature request: support for adding instrumentation/APM

### ISSUE THREAD

**Issue by bnkamalesh** [jackc/pgx#652]
Provision of adding instrumentation to `pgx/v4/pgxpool`. Right now, when using pgxpool, instrumentation has to be manually added, record errors etc. I glanced through the pgxpool sourcecode, though couldn't find any relevant interfaces or structs where I could plug in instrumentation. I think it would be awesome to have provision of adding it, as any production ready system at a decent scale would make use of monitoring. 

E.g. https://www.elastic.co/guide/en/apm/agent/go/current/instrumenting-source.html#builtin-modules-apmsql

**Comment by jackc** [comment]
Would wrapping the relevant pgxpool structs do what you want? pgxpool was specifically designed in a composable fashion. I suspect that a wrapper around pgxpool.Pool, pgxpool.Conn, and possibly pgxpool.Tx could do it.

**Comment by bnkamalesh** [comment]
@jackc Yes I believe it would help. I don't know if it already exists, I will have to spend some more time going through the code figure out. Thanks for the help!

P.S: it's an awesome package nevertheless

**Comment by jackc** [comment]
`v5` now has tracing support.
